iree/turbine/kernel/compiler/host_codegen.py

```diff
@@ -50,12 +51,11 @@ def isolated_test_call(
     dynamic_symbols: list[IndexSymbol] = [],
 ):
     with InsertionPoint(mb.body_block), Location.unknown():
-        input_types = [b.as_mlir_type() for b in sig.kernel_buffer_input_bindings]
+        input_types = [b.as_mlir_type() for b in sig.kernel_buffer_bindings]
         input_tensors = memref_to_tensor(input_types)
-        argument_dims = get_dynamic_dims(
-            sig.kernel_buffer_input_bindings, dynamic_symbols
-        )
-        input_tensors += [IndexType.get() for _ in argument_dims]
+        argument_dims = get_dynamic_dims(sig.kernel_buffer_bindings, dynamic_symbols)
+        # Adding unique dynamic dims as inputs.
+        input_tensors += [IndexType.get() for _ in list(dict.fromkeys(argument_dims))]
```

Comment:
This makes sense and is better than using set() since it preserves the insertion order.